adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -1119,6 +1119,12 @@ def _sock_exact_recv(self, bufsize: int) -> bytearray:
             rc = bytearray(bufsize)
             mv = memoryview(rc)
             recv_len = self._sock.recv_into(rc, bufsize)
+            if recv_len == 0:
```

Comment:
Is there some other case than timeout that could lead to the 0 being returned from `recv_into()`, regardless of the underlying implementation ? (ESP32, CPython, ...)

Reply:
Because if we are not sure about that, it might be prudent to augment the `recv_len == 0` check to verify this is actually esp32spi.

Reply:
That depends on the underlying implementation, so I wouldn't know. We could add check that receive_timeout seconds have passed before raising the exception, but that would require us to correctly define the receive timeout. As I mentioned earlier, even though the timeout passed to socket.settimeout is zero, my esp32spi socket waits a second before timing out.